File: src/main/java/com/skyspaceoneoneone/service/SeatService.java
package com.skyspaceoneoneone.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspaceoneoneone.domain.Seat;
import com.skyspaceoneoneone.dto.SeatDTO;
import com.skyspaceoneoneone.dto.SeatSearchDTO;
import com.skyspaceoneoneone.dto.SeatPageDTO;
import com.skyspaceoneoneone.dto.SeatConvertCriteriaDTO;
import com.skyspaceoneoneone.service.GenericService;
import com.skyspaceoneoneone.dto.common.RequestDTO;
import com.skyspaceoneoneone.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface SeatService extends GenericService<Seat, Integer> {

	List<Seat> findAll();

	ResultDTO addSeat(SeatDTO seatDTO, RequestDTO requestDTO);

	ResultDTO updateSeat(SeatDTO seatDTO, RequestDTO requestDTO);

    Page<Seat> getAllSeats(Pageable pageable);

    Page<Seat> getAllSeats(Specification<Seat> spec, Pageable pageable);

	ResponseEntity<SeatPageDTO> getSeats(SeatSearchDTO seatSearchDTO);
	
	List<SeatDTO> convertSeatsToSeatDTOs(List<Seat> seats, SeatConvertCriteriaDTO convertCriteria);

	SeatDTO getSeatDTOById(Integer seatId);







}

File: skyspaceoneoneone_ui/src/services/SeatService.js
import http from "../http-common"; 

class SeatService {
  getAllSeats(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/seat/seats`, searchDTO);
  }

  get(seatId) {
    return this.getRequest(`/seat/${seatId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/seat?field=${matchData}`, null);
  }

  addSeat(data) {
    return http.post("/seat/addSeat", data);
  }

  update(data) {
  	return http.post("/seat/updateSeat", data);
  }
  
  uploadImage(data,seatId) {
  	return http.postForm("/seat/uploadImage/"+seatId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new SeatService();
